app/models/hmpps_api/offender_base.rb

```diff
@@ -217,8 +217,16 @@ def within_early_allocation_window?
       earliest_date.present? && earliest_date <= Time.zone.today + 18.months
     end
 
+    def needs_com_and_ldu_is_uncontactable?
+      needs_com? && sentenced? && allocated_com_name.nil? && ldu_email_address.nil?
+    end
+
   private
 
+    def needs_com?
+      handover.community.responsible? || handover.community.supporting?
+    end
```

Comment:
The method names here are slightly misleading:

- `needs_com?` = should a COM be involved in the case?
- `needs_com_and_ldu_is_uncontactable?` = same as `needs_com?` **but also checks if a COM is already allocated** – as well as if the LDU is contactable

Given that they share the beginning of their name, it seems a bit weird that the addition of `_and_ldu_is_uncontactable` to the method name **also adds a check for COM allocation**.

It's the sort of subtle difference that might catch someone out.

However: I recognise the need for both methods, and haven't yet come up with a better name. So just something to think about.